cap/modules/deposit/utils.py

```diff
@@ -93,3 +96,24 @@ def prepare_record(sender, json=None, record=None,
 
     collection = {"name": name, "version": version, "fullname": fullname}
     json["_collection"] = collection
+
+
+def generate_auto_incremental_pid():
+    """Method to generate auto-increment PID for records/deposits.
+
+    - Enabled for FASER experiment.
+    """
+    try:
+        previous_deposit_id = 0
+        previous_deposit = PersistentIdentifier.query.filter(
+            PersistentIdentifier.pid_value.contains('FASER'),
+            PersistentIdentifier.pid_type == 'depid'
+        ).order_by(PersistentIdentifier.id.desc()).first()
```

Comment:
shouldn't it be ordered by `pid_value`? like this what would happen if other records from different schemas are submitted between 2 FASER ones?

Reply:
Here, I get the last FASER deposit created. The ordering happens after filtering so other records will not be there while ordering the records and getting the last one. Tested it.